File: python/leetcode/012.py
class Solution:
    def intToRoman(self, num):
        if num < 0:
            raise ValueError("negative num")
        romans = ['I', 'IV', 'V', 'IX', 'X', 'XL', 'L', 'XC', 'C', 'CD', 'D', 'CM', 'M']
        nums = [1, 4, 5, 9, 10, 40, 50, 90, 100, 400, 500, 900, 1000]
        res = []
        for i in range(len(nums) - 1, -1, -1):
            while num >= nums[i]:
                res.append(romans[i])
                num -= nums[i]
        return "".join(res)

File: cpp/leetcode/012.cpp
class Solution {
  public:
    string intToRoman(int num) {
        if (num < 0) {
            throw invalid_argument("negative number");
        }
        string romans[13] = {"I", "IV", "V", "IX", "X", "XL", "L", "XC", "C", "CD", "D", "CM", "M"};
        int nums[13] = {1, 4, 5, 9, 10, 40, 50, 90, 100, 400, 500, 900, 1000};
        string res = "";
        for (int i = 12; i >= 0; --i) {
            while (num >= nums[i]) {
                res += romans[i];
                num -= nums[i];
            }
        }
        return res;
    }
};